### PR TITLE
Add IteratorAsText and IterableAsText

### PR DESCRIPTION
**Pull request by smallcreep**
As per #135

### CODE REVIEW

**Comment by 0crat** [comment]
@smallcreep I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/list/IteratorAsText.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        final Text separator
+    ) {
+        this.iterator = iterator;
+        this.pattern = ptn;
+        this.separator = separator;
+    }
+
+    @Override
+    public String asString() throws IOException {
+        final StringBuilder builder = new StringBuilder();
```
Comment (Timmeey) [inline src/main/java/org/cactoos/list/IteratorAsText.java]: this should use `JoinedText` to join texts instead of stringbuilder i think.

**Comment by yegor256** [comment]
@smallcreep yes, we have to use `JoinedText`

**Comment by smallcreep** [comment]
@Timmeey @yegor256 I added ```JoinedText```, and use it in ```IteratorAsText```

File: src/main/java/org/cactoos/list/IteratorAsText.java (hunk 1 above)
Comment (smallcreep) [inline src/main/java/org/cactoos/list/IteratorAsText.java]: @Timmeey @yegor256 I added JoinedText, and use it in IteratorAsText

**Pull request closed without merging**